src/AppBundle/Security/FormLoginAuthenticator.php

```diff
@@ -1,91 +1,99 @@
-use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
+use Symfony\Component\Security\Core\Encoder\UserPasswordEncoder;
 use Symfony\Component\Security\Core\Exception\BadCredentialsException;
+use Symfony\Component\Security\Guard\Authenticator\AbstractFormLoginAuthenticator;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
 use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\Security;
 use Symfony\Component\Security\Core\User\UserInterface;
 use Symfony\Component\Security\Core\User\UserProviderInterface;
-use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
-use Symfony\Component\Security\Guard\Authenticator\AbstractFormLoginAuthenticator;
-use Symfony\Component\Security\Core\Security;
 
 class FormLoginAuthenticator extends AbstractFormLoginAuthenticator
 {
+    private $em;
     private $router;
-    private $encoder;
+    private $passwordEncoder;
 
-    public function __construct(RouterInterface $router, UserPasswordEncoderInterface $encoder)
+    public function __construct(EntityManager $em, RouterInterface $router, UserPasswordEncoder $passwordEncoder)
     {
+        $this->em = $em;
         $this->router = $router;
-        $this->encoder = $encoder;
+        $this->passwordEncoder = $passwordEncoder;
     }
 
     public function getCredentials(Request $request)
     {
-        if ($request->getPathInfo() != '/login_check') {
-          return;
+// if getCredentials() returns null, the authenticator is skipped.
+        if ($request->getPathInfo() != '/login' || !$request->isMethod('POST')) {
+            return null;
         }
 
-        $email = $request->request->get('_email');
-        $request->getSession()->set(Security::LAST_USERNAME, $email);
+        $username = $request->request->get('_userName');
+        $request->getSession()->set(Security::LAST_USERNAME, $username);
         $password = $request->request->get('_password');
 
-        return [
-            'email' => $email,
-            'password' => $password,
-        ];
+        return array(
+            'username' => $username,
+            'password' => $password
+        );
     }
 
     public function getUser($credentials, UserProviderInterface $userProvider)
     {
-        $email = $credentials['email'];
+        $username = $credentials['username'];
+
+        $user = $this->em->getRepository('AppBundle:User')->findOneBy(['email' => $username]);
+
+        if (!$user) {
+            throw new BadCredentialsException();
+        }
 
-        return $userProvider->loadUserByUsername($email);
+        return $user;
     }
 
     public function checkCredentials($credentials, UserInterface $user)
     {
         $plainPassword = $credentials['password'];
-        if ($this->encoder->isPasswordValid($user, $plainPassword)) {
+
+        if ($this->passwordEncoder->isPasswordValid($user, $plainPassword)) {
```

Comment:
logiškiau būtų grąžinti tik bool reikšmę pagal metodo pavadinimą `return $this->passwordEncoder->isPasswordValid($user, $plainPassword);`